### PR TITLE
Add 3.2.1 release note and news and update links 

### PR DESCRIPTION
**Pull request by huaxingao**
This PR aims to update our website. I checked the followings are available as of now.

https://spark.apache.org/docs/3.2.1/
https://dist.apache.org/repos/dist/release/spark/spark-3.2.1/
https://repo1.maven.org/maven2/org/apache/spark/spark-core_2.12/3.2.1/
https://pypi.org/project/pyspark/3.2.1/

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @gengliangwang @dongjoon-hyun @viirya

File: site/documentation.html [hunk 1]
```diff
@@ -149,6 +149,7 @@ <h2><span class="text-capitalize">Apache Spark<span class="tm">&trade;</span></s
 <p>Setup instructions, programming guides, and other documentation are available for each stable version of Spark below:</p>
 
 <ul>
+  <li><a href="/docs/3.2.0/">Spark 3.2.1</a></li>
```
Comment (srowen) [inline site/documentation.html]: (Should be 3.2.1)

**Review by srowen — CHANGES_REQUESTED** [review]

File: site/documentation.html (hunk 1 above)
Comment (viirya) [reply site/documentation.html]: I think this should be generated by `documentation.md`? @huaxingao do you manually modify this file?

Comment (huaxingao) [reply site/documentation.html]: I had the wrong version in `documentation.md`. I fixed it but forgot to run `jekyll build`.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
@huaxingao FYI it would be nice if we can announce the release after https://github.com/algolia/docsearch-configs/pull/5054 is merged. Otherwise, the search box of the Spark doc site won't work.
This is a limitation in current days. (Of course, we can see how to improve the process in the future)

**Comment by huaxingao** [comment]
OK. Thanks for reviewing! @gengliangwang

File: js/downloads.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ var packagesV11 = [hadoop3p2, hadoop2p7, hadoopFree, sources];
 // 3.2.0+
 var packagesV12 = [hadoop3p3, hadoop3p3scala213, hadoop2p7, hadoopFree, sources];
 
+addRelease("3.2.1", new Date("01/26/2022"), packagesV12, true);
 addRelease("3.2.0", new Date("10/13/2021"), packagesV12, true);
```
Comment (dongjoon-hyun) [inline js/downloads.js]: We should remove this at this time because this is superseded by 3.2.1.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM with one minor comment about downloading page and also +1 for @gengliangwang 's comment.

**Review by viirya — APPROVED** [review]

**Comment by huaxingao** [comment]
@srowen Hi Sean, thanks for catching the problem! If you don't have any more concerns, I will merge?

**Review by srowen — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @huaxingao and all!

**Comment by huaxingao** [comment]
Thank you all very much for your help!

**Comment by dongjoon-hyun** [comment]
To @gengliangwang , according to your comment, is it still not working with unknown reasons?

https://github.com/algolia/docsearch-configs/pull/5054#issuecomment-1022872245

Since they changed it to a self-service mode and we don't have anything to do further there, I believe we need to proceed the Apache Spark 3.2.1 announcement without waiting for it. ASF release should not be blocked by 3rd-party service's outage or bugs.

Please proceed to the announcement, @huaxingao .

**Comment by gengliangwang** [comment]
@dongjoon-hyun @huaxingao yes, I will manage to fix it later.

**Pull request merged**